digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/ServicePointBusStopDao.scala

```diff
@@ -0,0 +1,262 @@
+    val rows = Q.queryNA[ServicePointRow](query).iterator.toSeq
+
+    rows.groupBy(_.id).map { case (id, stopRows) =>
+      val row = stopRows.head
+      val commonProperties: Seq[Property] = AssetPropertyConfiguration.assetRowToCommonProperties(row)
+      val properties: Seq[Property] = commonProperties ++ assetRowToProperty(stopRows)
+      val point = row.point.get
+      val stopTypes = extractStopTypes(stopRows)
+
+      id -> ServicePoint(id = row.id, nationalId = row.externalId, stopTypes = stopTypes,
+        municipalityCode = row.municipalityCode, lon = point.x, lat = point.y, created = row.created, modified = row.modified, propertyData = properties)
+    }.values.toSeq
+  }
+
+  private def extractStopTypes(rows: Seq[ServicePointRow]): Seq[Int] = {
+    rows.filter(_.property.publicId.equals(MassTransitStopOperations.MassTransitStopTypePublicId)).map { row => row.property.propertyValue.toInt }
+  }
+
+  def expire(id: Long, username: String) = {
+    Queries.updateAssetModified(id, username).execute
+    sqlu"update asset set valid_to = sysdate where id = $id".execute
+  }
+
+  def update(assetId: Long, properties: Seq[SimplePointAssetProperty], user: String) = {
+    sqlu"""
+           UPDATE asset
+            SET modified_by = $user, modified_date = sysdate
+            WHERE id = $assetId""".execute
+
+    updateAssetProperties(assetId, properties)
+    assetId
+  }
+
+  def insertAsset(id: Long, lon: Double, lat: Double, creator: String, municipalityCode: Int): Unit = {
+    sqlu"""insert into asset (id, external_id, asset_type_id, created_by, municipality_code, geometry)
+           select $id, national_bus_stop_id_seq.nextval, $typeId, $creator, $municipalityCode,
+           MDSYS.SDO_GEOMETRY(4401, 3067, NULL, MDSYS.SDO_ELEM_INFO_ARRAY(1,1,1), MDSYS.SDO_ORDINATE_ARRAY($lon, $lat, 0, 0))
+           from dual
+      """.execute
+  }
+
+  def updateMunicipality(id: Long, municipalityCode: Int) {
+    sqlu"""
+           update asset
+           set municipality_code = $municipalityCode
+           where id = $id
+      """.execute
+  }
+
+  private[this] def createOrUpdateMultipleChoiceProperty(propertyValues: Seq[PropertyValue], assetId: Long, propertyId: Long) {
+    val newValues = propertyValues.map(_.propertyValue.toLong)
+    val currentIdsAndValues = Q.query[(Long, Long), (Long, Long)](multipleChoicePropertyValuesByAssetIdAndPropertyId).apply(assetId, propertyId).list
+    val currentValues = currentIdsAndValues.map(_._2)
+    // remove values as necessary
+    currentIdsAndValues.foreach {
+      case (multipleChoiceId, enumValue) =>
+        if (!newValues.contains(enumValue)) {
+          deleteMultipleChoiceValue(multipleChoiceId).execute
+        }
+    }
+    // add values as necessary
+    newValues.filter {
+      !currentValues.contains(_)
+    }.foreach {
+      v =>
+        insertMultipleChoiceValue(assetId, propertyId, v).execute
+    }
+  }
```

Comment:
Identical on MassTransitStopDao class

Reply:
Extending MassTransitStopDao seems to be the best option

Reply:
@marquesfm This part was removed since this dao now extends from the masstransitstop one.
Check modification on code of the def on the masstransitstop dao (it is now protected)